### PR TITLE
Update Azure CLI quantum extension to multi-region

### PR DESCRIPTION
**Pull request by ricardo-espinoza**
This change will set the URL for the data plan API accordingly with a location parameter specified as part of the command.

### CODE REVIEW

File: src/quantum/azext_quantum/_location_helper.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import re
+
+DEFAULT_WORKSPACE_LOCATION = 'westus'
+
+# Currently, we're only checking that the provided location doesn't contain unsafe characters
+# but there is no guarantee that the returned value exists as an Azure region.
+# If an invalid region is specified, then the error will happen when the corresponding API
+# endpoint isn't found.
+def normalize_location(raw_location):
+    if (not raw_location):
```
Comment (rmshaffer) [inline src/quantum/azext_quantum/_location_helper.py]: nit, unnecessary parens:
```suggestion
    if not raw_location:
```

File: src/quantum/azext_quantum/_location_helper.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        return DEFAULT_WORKSPACE_LOCATION
+    location = re.sub("[^A-Za-z0-9]","",raw_location).lower()
+    if (location == ""):
```
Comment (rmshaffer) [inline src/quantum/azext_quantum/_location_helper.py]: nit:
```suggestion
    if not location:
```

File: src/quantum/azext_quantum/_validators.py [hunk 3]
```diff
@@ -8,21 +8,36 @@
...
+def validate_workspace_info(cmd, namespace):
+    """
+    Makes sure all parameters for a workspace are available including location.
+    """
+    validate_workspace_internal(cmd, namespace, True)
+
+
+def validate_workspace_info_no_location(cmd, namespace):
+    """
+    Makes sure all parameters for a workspace are available including location.
```
Comment (rmshaffer) [inline src/quantum/azext_quantum/_validators.py]:
```suggestion
    Makes sure all parameters for a workspace are available, not including location.
```

File: src/quantum/azext_quantum/_location_helper.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import re
+
+DEFAULT_WORKSPACE_LOCATION = 'westus'
+
+# Currently, we're only checking that the provided location doesn't contain unsafe characters
+# but there is no guarantee that the returned value exists as an Azure region.
+# If an invalid region is specified, then the error will happen when the corresponding API
+# endpoint isn't found.
+def normalize_location(raw_location):
```
Comment (rmshaffer) [inline src/quantum/azext_quantum/_location_helper.py:14]: Would it be worth adding tests for valid/invalid locations?

**Review by rmshaffer — APPROVED** [review]

File: src/quantum/azext_quantum/_location_helper.py (hunk 1 above)
Comment (ricardo-espinoza) [reply src/quantum/azext_quantum/_location_helper.py]: C++ habits die hard...

File: src/quantum/azext_quantum/_location_helper.py (hunk 4 above)
Comment (ricardo-espinoza) [reply src/quantum/azext_quantum/_location_helper.py:14]: Makes sense, I'll add one in the next round to avoid resetting this change.

**Pull request merged**